internal/dataplane/translator/ingressrules.go

```diff
@@ -21,7 +21,13 @@ import (
 	"github.com/kong/kubernetes-ingress-controller/v3/internal/util"
 )
 
-func getClientCertIncompatibleProtocols() []string {
+const (
+	// defaultServiceProtocol is the default protocol used by Kong for services.
+	defaultServiceProtocol = "http"
+)
+
+// getNonTLSProtocols returns a list of protocols that do not make Kong establish a TLS connection with an upstream.
+func getNonTLSProtocols() []string {
```

Comment:
It returns to use it with `lo.Contains(...)`, maybe it can be `isNonTLSProtocol(p string) bool` and used like that later?

Reply:
Yeah, that makes sense 👍 PTAL https://github.com/Kong/kubernetes-ingress-controller/pull/6707/commits/74ccc7a96154eb0e42c02a63149b03b2857193c1